src/pudl/transform/ferc1.py

```diff
@@ -3866,3 +3885,474 @@ def plants_steam_ferc1(
+        )
+    return meta_converted
+
+
+def check_table_calcs(
+    table_name: str,
+    table_df: pd.DataFrame,
+    dollar_value_col: str,
+    meta_converted: dict,
+):
+    """Check how well a table's calculated values are calculated."""
+    xbrl_factoid_name = FERC1_TFR_CLASSES[table_name]().params.merge_xbrl_metadata.on
+    pks = (
+        pudl.metadata.classes.Package.from_resource_ids()
+        .get_resource(table_name)
+        .schema.primary_key
+    )
+    pks_wo_factoid = [col for col in pks if col != xbrl_factoid_name]
+    meta_converted = ensure_names_in_renamed_xbrl_calcs_are_present(
+        df=table_df,
+        meta_converted=meta_converted,
+        xbrl_factoid_name=xbrl_factoid_name,
+        table_name=table_name,
+    )
+    inter_table_calculated_values = {
+        field: calc_components
+        for field, field_info in meta_converted[table_name].items()
+        if field_info.get("calcs", False)
+        for calc_components in field_info["calcs"]
+        if calc_components.get("source_table", False)
+    }
+    intra_table_calculated_values = {
+        field: calc_components
+        for field, field_info in meta_converted[table_name].items()
+        if field_info.get("calcs", False)
+        for calc_components in field_info["calcs"]
+        if not calc_components.get("source_table", False)
+    }
+    if inter_table_calculated_values:
+        logger.info(
+            "Skipping calcuated values because they are inter-table calucations: "
+            f"{inter_table_calculated_values.keys()}"
+        )
+    calculated_dfs = []
+    for cv in intra_table_calculated_values:
+        df = (
+            pd.merge(
+                table_df,
+                pd.DataFrame(meta_converted[table_name][cv]["calcs"]),
+                left_on=xbrl_factoid_name,
+                right_on="name",
+            )
+            .assign(calculated_dollar_amount=lambda x: x[dollar_value_col] * x.weight)
+            .groupby(pks_wo_factoid)
+            .calculated_dollar_amount.sum()
+            .to_frame()
+            .reset_index()
+        )
+        df[xbrl_factoid_name] = cv
+        calculated_dfs.append(df)
+
+    calculated_df = pd.merge(
+        table_df,
+        pd.concat(calculated_dfs),
+        on=pks,
+        how="left",
+    )
+
+    calculated_df["abs_diff"] = abs(
+        calculated_df[dollar_value_col] - calculated_df.calculated_dollar_amount
+    )
+    calculated_df["rel_diff"] = abs(
+        calculated_df.abs_diff / calculated_df[dollar_value_col]
+    )
+    off_df = calculated_df[
+        ~np.isclose(
+            calculated_df.calculated_dollar_amount, calculated_df[dollar_value_col]
+        )
+        & (calculated_df["abs_diff"].notnull())
+    ]
+    calced_values = calculated_df[(calculated_df.abs_diff.notnull())]
+    logger.info(
+        f"{table_name}: has #{len(off_df)} / {len(off_df)/len(calced_values):.02%} records that don't calculate exactly"
```

Comment:
I found this logging message confusing since it looks like division. I think "table_name has 78 records (2.35%) that don't calculate exactly." would be clearer to me.